packages/eas-cli/src/commands/update/index.ts

```diff
@@ -600,15 +630,69 @@ async function getRuntimeVersionObjectAsync(
     }
   }
 
-  return Object.fromEntries(
-    platforms.map(platform => [
-      platform,
-      nullthrows(
-        Updates.getRuntimeVersion(exp, platform),
-        `Unable to determine runtime version for ${platform}`
-      ),
-    ])
-  );
+  try {
+    return [transformRuntimeVersions(exp, platforms), exp];
+  } catch (error: any) {
+    if (nonInteractive) {
+      throw error;
+    }
+
+    Log.fail(error.message);
+
+    const runConfig = await selectAsync(
+      `Do you want us to run automatic ${chalk.bold(
+        'app.json'
+      )} configuration for EAS Update for you?`,
+      [
+        { title: 'Yes', value: true },
+        {
+          title: 'No, I will set the runtime version manually (EAS CLI exits)',
+          value: false,
+        },
+      ]
+    );
+
+    if (!runConfig) {
+      Errors.exit(1);
+    }
+
+    const workflows = await resolveWorkflowPerPlatformAsync(projectDir);
+    const configUpdate = await configureAppJSONForEASUpdateAsync({
+      exp,
+      projectDir,
+      projectId,
+      platform: platformFlag as RequestedPlatform,
+      workflows,
+    });
+
+    const newConfig: ExpoConfig = { ...exp, ...configUpdate };
+
+    await configureNativeFilesForEASUpdateAsync({
+      exp: newConfig,
+      projectDir,
+      projectId,
+      platform: platformFlag as RequestedPlatform,
+      workflows,
+      graphqlClient,
+    });
```

Comment:
Based on the code in `configure`, I think the result of `configureAppJSONForEASUpdateAsync` is the source of truth for `newConfig`.

```suggestion
    const newConfig = await configureAppJSONForEASUpdateAsync({
      exp,
      projectDir,
      projectId,
      platform: platformFlag as RequestedPlatform,
      workflows,
    });

    await configureNativeFilesForEASUpdateAsync({
      exp: newConfig,
      projectDir,
      projectId,
      platform: platformFlag as RequestedPlatform,
      workflows,
      graphqlClient,
    });
```

Reply:
@wschurman I have stated the reasoning for how this code works in conversation with Jon (should post a comment here probably), let me quote it:

> the problem was that `configureAppJSONForEASUpdateAsync` returns updated config, but it was only a partial with the changes, so instead of using result directly we need to merge it with full config

It was a root cause why `"Error: Unable to determine runtime version for android"` started appearing after one of the PR changes. The problem has been mentioned in this comment:
* https://github.com/expo/eas-cli/pull/1427#pullrequestreview-1144562679

Reply:
Looking at `modifyConfigAsync` source, it looks like it returns exactly what's on disk? Maybe we need to run `transformRuntimeVersions` again or something?

Reply:
I don't think that `modifyConfigAsync` is a source of problem there, it already receives config partial:

* https://github.com/expo/eas-cli/blob/b9c505bbdc30af90b17633028c171465e9ec8df2/packages/eas-cli/src/commands/update/configure.ts#L128-L129

Reply:
Ah, I see. In that case we should re-fetch the full config using `getDynamicProjectConfigAsync`. Doing just an object merge will overwrite keys (doesn't do a deep merge).

```suggestion
    const configUpdate = await configureAppJSONForEASUpdateAsync({
      exp,
      projectDir,
      projectId,
      platform: platformFlag as RequestedPlatform,
      workflows,
    });

    const newConfig = (await getDynamicProjectConfigAsync({ isPublicConfig: true })).exp;

    await configureNativeFilesForEASUpdateAsync({
      exp: newConfig,
      projectDir,
      projectId,
      platform: platformFlag as RequestedPlatform,
      workflows,
      graphqlClient,
    });
```